### PR TITLE
Clean, e2e and testing

### PR DESCRIPTION
**Pull request by NicholasNagy**
This pull request includes the following:

E2E tests for the remaining user stories.
Front-end tests, bringing our front end coverage to about 69.96%! Nice!
Clean up of the code smells and duplications identified by sonar cloud.

This will be our final PR, so before merging, we should really take the time to test everything out to make sure everything is working for all user stories.

### CODE REVIEW

File: client/src/app/components/employees/employee-view/edit-employee/edit-employee.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  // it('edit mode', fakeAsync(() => {
+  //   try {
+  //    // const spy = spyOn(manageMembersService, 'getEmployee').and.returnValue();
+  //
+  //     component.editMode(false);
+  //   } catch (e) {
+  //     console.error(e);
+  //   }
+  // }));
+});
```
Comment (makchamp) [inline client/src/app/components/employees/employee-view/edit-employee/edit-employee.component.spec.ts:135]: can this be deleted

File: client/src/app/components/manage-audits/manage-audits.component.ts [hunk 2]
```diff
@@ -414,12 +407,10 @@ export class ManageAuditsComponent extends TableManagementComponent implements O
 
     this.items.forEach((audit: any) => {
       audit.isSelected = false;
-
-    }  );
+    });
 
     // @ts-ignore
     this.dataSource = new MatTableDataSource(this.items);
-    // console.log(this.items)
```
Comment (Amanda-Kolopa) [inline client/src/app/components/manage-audits/manage-audits.component.ts:422]: Can we remove this?

File: client/src/app/components/audit-report/comment/comment.component.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import {AfterViewChecked, Component, OnInit} from '@angular/core';
-import {AuditReportService} from "../../../services/audits/audit-report.service";
-import {ManageMembersService} from "../../../services/users/manage-members.service";
+import {AuditReportService} from '../../../services/audits/audit-report.service';
+import {ManageMembersService} from '../../../services/users/manage-members.service';
```
Comment (makchamp) [inline client/src/app/components/audit-report/comment/comment.component.ts:3]: we can avoid relative imports

File: client/src/app/components/employees/employee-view/edit-employee/edit-employee.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+import {TokenService} from 'src/app/services/authentication/token.service';
+import {HttpClientTestingModule} from '@angular/common/http/testing';
+import {RouterTestingModule} from '@angular/router/testing';
+import {EditEmployeeComponent} from './edit-employee.component';
+import {FormBuilder} from '@angular/forms';
+import {ManageOrganizationsService} from 'src/app/services/organizations/manage-organizations.service';
+import 'zone.js/dist/zone-testing';
+import {AppModule} from 'src/app/app.module';
+import {of, throwError} from 'rxjs';
+import {ManageMembersService} from '../../../../services/users/manage-members.service';
```
Comment (makchamp) [inline client/src/app/components/employees/employee-view/edit-employee/edit-employee.component.spec.ts:12]: we can avoid relative imports

File: client/src/app/services/authentication/auth.service.spec.ts [hunk 5]
```diff
@@ -91,4 +95,96 @@ describe('AuthService', () => {
...
+  it('test turnOffOrgMode method', () => {
+    spyOn(authService, 'removeFromLocalStorage').and.callFake(() => {});
+    authService.turnOffOrgMode();
+  });
+
+  // turnOffOrgMode
+  it('test turnOffOrgMode method', () => {
+    spyOn(authService, 'removeFromLocalStorage').and.callFake(() => {});
+    authService.turnOffOrgMode();
+  });
```
Comment (makchamp) [inline client/src/app/services/authentication/auth.service.spec.ts:128]: duplicate tests?

File: client/src/app/services/authentication/auth.service.spec.ts [hunk 6]
```diff
@@ -91,4 +95,96 @@ describe('AuthService', () => {
...
+
+  // OnDestroy
+  it('test OnDestroy method', () => {
+    authService.OnDestroy();
+  });
+
+  // getCurrentUser
+  it('test getCurrentUser method', () => {
+    let userId = '7';
+    // let updateInfo = {first_name: 'a good name', last_name: "admin"};
```
Comment (makchamp) [inline client/src/app/services/authentication/auth.service.spec.ts:160]: can we remove this coment

File: client/src/app/services/authentication/auth.service.spec.ts [hunk 7]
```diff
@@ -91,4 +95,96 @@ describe('AuthService', () => {
...
+      expect(data).toEqual({});
+    });
+
+    const req = httpTestingController.expectOne(`${BASEURL}/open-registration/`);
+    expect(req.request.method).toBe('POST');
+  });
+
+  // openRegister
+  it('test loginMobile method', () => {
+    // let updateInfo = {first_name: 'a good name', last_name: "admin"};
```
Comment (makchamp) [inline client/src/app/services/authentication/auth.service.spec.ts:182]: can we remove this coment

**Review by makchamp — CHANGES_REQUESTED** [review]
Thanks for the extensive testing guys. I just had a few minor comments to resolve.

**Comment by NicholasNagy** [comment]
thanks @makchamp , I updated the PR based on your review

**Review by makchamp — APPROVED** [review]

**Pull request merged**